ami/trainers/i_jepa_latent_visualization_decoder_trainer.py

```diff
@@ -0,0 +1,167 @@
+# Ref: https://github.com/facebookresearch/ijepa
+
+from functools import partial
+from pathlib import Path
+from typing import Literal
+
+import torch
+import torchvision
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+from typing_extensions import override
+
+from ami.data.buffers.buffer_names import BufferNames
+from ami.data.buffers.random_data_buffer import RandomDataBuffer
+from ami.data.interfaces import ThreadSafeDataUser
+from ami.models.i_jepa import IJEPAEncoder
+from ami.models.i_jepa_latent_visualization_decoder import (
+    IJEPALatentVisualizationDecoder,
+)
+from ami.models.model_names import ModelNames
+from ami.models.model_wrapper import ModelWrapper
+from ami.tensorboard_loggers import StepIntervalLogger
+
+from .base_trainer import BaseTrainer
+
+
+class IJEPALatentVisualizationDecoderTrainer(BaseTrainer):
+    def __init__(
+        self,
+        partial_dataloader: partial[DataLoader[torch.Tensor]],
+        partial_optimizer: partial[Optimizer],
+        device: torch.device,
+        logger: StepIntervalLogger,
+        encoder_type: Literal[ModelNames.I_JEPA_TARGET_ENCODER, ModelNames.I_JEPA_CONTEXT_ENCODER],
```

Comment:
「型」ではなく「名前」ので`type`ではなく`name`にしよう。
あと、decoder_nameを指定しよう。一つのEncoderに対応する Decoderを学習するという風にするとわかりやすくなる。
```suggestion
        encoder_name: Literal[ModelNames.I_JEPA_TARGET_ENCODER, ModelNames.I_JEPA_CONTEXT_ENCODER],
        decoder_name: Literal[ModelNames.I_JEPA_TARGET_DECODER, ModelNames.I_JEPA_CONTEXT_DECODER]
```